assets/src/application/services/apollo/relations/reducer.ts

```diff
@@ -0,0 +1,78 @@
+import { pathOr, assocPath, dissocPath, clone } from 'ramda';
+import { RelationsReducer, RelationalData } from './types';
+
+const reducer: RelationsReducer = (state, action) => {
+	const { entity, entityId, relation, relationId, relationIds } = action;
+	let newState: RelationalData, relations: string[];
+	switch (action.type) {
+		case 'INITIALIZE':
+			return action.data;
+
+		case 'ADD_RELATION':
+			relations = pathOr([], [entity, entityId, relation], state);
+			// If the relation already exists
+			if (relations.includes(relationId)) {
+				return state;
+			}
+			newState = assocPath([entity, entityId, relation], [...relations, relationId], state);
+			return newState;
+
+		case 'REMOVE_RELATION':
+			newState = clone(state);
+			// existing relation list.
+			relations = pathOr([], [entity, entityId, relation], newState);
```

Comment:
good candidate for `nullish coalescing operator` :D :D

Reply:
That will be like this
```suggestion
			relations = newState?.[entity]?.[entityId]?.[relation] ?? [];
```